uber/models.py

```diff
@@ -593,27 +593,12 @@ def auto_badge_num(self, badge_type):
 
                 # Searches badge range for a gap in badge numbers; if none found, returns the latest badge number + 1
                 # Doing this lets admins manually set high badge numbers without filling up the badge type's range.
-                first = 0
-                last = sametype.count() - 1
-                if sametype_list[last] - sametype_list[first] == last:
+                start, end = c.BADGE_RANGES[badge_type][0], sametype_list[-1]
+                gap_nums = sorted(set(range(start, end + 1)).difference(sametype_list))
```

Comment:
No, that isn't what I was thinking. What I was thinking was:
For a badge_range whose first # is defined to be 1000 and last # is defined to be 1999, where the first assigned badge is 1000 and the last assigned badge is 1499, wouldn't gap_nums contain 1500, 1501, ... 1999?
For that matter, what happens when the badge_range is full (i.e. last assigned badge is 1999)? gap_nums would actually be empty, and you'd attempt to assign a badge number out of range?